### ISSUE TITLE
Warning when testing method "base-repeating" (possible DESTROY issue)

### ISSUE THREAD

**Issue by jubilatious1** [rakudo/rakudo#3498]
## The Problem

REPL throws warning during test of "method base-repeating" code taken from this page:  
https://docs.raku.org/type/Rational#method_base-repeating

Warning suggests old value is not destroyed when new value is entered:
```
> say (1/7).base-repeating(10);
(0. 142857)
> say (1/7).base-repeating(100);
base argument to base out of range. Is: 100, should be in 2..36
  in block <unit> at <unknown file> line 1

> say (1/7).base-repeating(36);
(0. 5)
> say (1/7).base-repeating(2);
WARNING: unhandled Failure detected in DESTROY. If you meant to ignore it, you can mark it as handled by calling .Bool, .so, .not, or .defined methods. The Failure was:
base argument to base out of range. Is: 100, should be in 2..36
  in block <unit> at <unknown file> line 1

WARNING: unhandled Failure detected in DESTROY. If you meant to ignore it, you can mark it as handled by calling .Bool, .so, .not, or .defined methods. The Failure was:
base argument to base out of range. Is: 100, should be in 2..36
  in block <unit> at <unknown file> line 1

WARNING: unhandled Failure detected in DESTROY. If you meant to ignore it, you can mark it as handled by calling .Bool, .so, .not, or .defined methods. The Failure was:
base argument to base out of range. Is: 100, should be in 2..36
  in block <unit> at <unknown file> line 1

(0. 001)
> say (1/7).base-repeating(36);
(0. 5)

```
## Expected Behavior / Actual Behavior

When the code `say (1/7).base-repeating(2);` is entered, I expected the REPL to run using the "2" parameter/value that had been passed in, inside the parentheses. 

However the REPL reported a 'DESTROY' warning suggesting that it was still using a value of "100" that had been passed in, inside the parentheses two entries earlier.

## Environment
* Operating system: MacOSX
* Compiler version (`perl6 -v`): This is Rakudo version 2019.07.1 built on MoarVM version 2019.07.1 implementing Perl 6.d.

**Comment by jubilatious1** [comment]
I probably should mention that upon quitting that REPL session, I received 6 warnings. [Six warnings possibly attributable to calling `say (1/7).base-repeating(100);` twice and accumulating three warnings each time]. See below:

```
> say (1/7).base-repeating(2);
(0. 001)
> say (1/7).base-repeating(10);
(0. 142857)
> say (1/7).base-repeating(100);
base argument to base out of range. Is: 100, should be in 2..36
  in block <unit> at <unknown file> line 1

> say (1/7).base-repeating(2);
(0. 001)
> say (1/7).base-repeating(100);
base argument to base out of range. Is: 100, should be in 2..36
  in block <unit> at <unknown file> line 1

> say (1/7).base-repeating(36);
(0. 5)
> exit
WARNING: unhandled Failure detected in DESTROY. If you meant to ignore it, you can mark it as handled by calling .Bool, .so, .not, or .defined methods. The Failure was:
base argument to base out of range. Is: 100, should be in 2..36
  in block <unit> at <unknown file> line 1

WARNING: unhandled Failure detected in DESTROY. If you meant to ignore it, you can mark it as handled by calling .Bool, .so, .not, or .defined methods. The Failure was:
base argument to base out of range. Is: 100, should be in 2..36
  in block <unit> at <unknown file> line 1

WARNING: unhandled Failure detected in DESTROY. If you meant to ignore it, you can mark it as handled by calling .Bool, .so, .not, or .defined methods. The Failure was:
base argument to base out of range. Is: 100, should be in 2..36
  in block <unit> at <unknown file> line 1

WARNING: unhandled Failure detected in DESTROY. If you meant to ignore it, you can mark it as handled by calling .Bool, .so, .not, or .defined methods. The Failure was:
base argument to base out of range. Is: 100, should be in 2..36
  in block <unit> at <unknown file> line 1

WARNING: unhandled Failure detected in DESTROY. If you meant to ignore it, you can mark it as handled by calling .Bool, .so, .not, or .defined methods. The Failure was:
base argument to base out of range. Is: 100, should be in 2..36
  in block <unit> at <unknown file> line 1

WARNING: unhandled Failure detected in DESTROY. If you meant to ignore it, you can mark it as handled by calling .Bool, .so, .not, or .defined methods. The Failure was:
base argument to base out of range. Is: 100, should be in 2..36
  in block <unit> at <unknown file> line 1

mbook:~ homedir$

```

**Comment by jubilatious1** [comment]
This issue appears to be resolved in Rakudo version 2020.02.1:

```
mbook:~ homedir$ perl6 -v
This is Rakudo version 2020.02.1.0000.1 built on MoarVM version 2020.02.1
implementing Raku 6.d.
```
